### PR TITLE
Update Dependencies and move to Gradle 8.1 (Kotlin 1.8.10)

### PR DESCRIPTION
**Pull request by JT122406**

### CODE REVIEW

**Comment by modmuss50** [comment]
Hi, thanks for the PR. For the moment updating Gradle like this is not something I want to do just yet. My plan was to move to Gradle 8.1 with the next loom version as there are some neat features we can make use of. This will also require bumping the min required Gradle version in the bootstrap.

Ill leave this open to come back to it when I start work on the next version of loom.

**Comment by JT122406** [comment]
sounds good!

File: build.gradle [hunk 1]
```diff
@@ -95,26 +95,26 @@ dependencies {
 	implementation ('net.fabricmc:mercury:0.3.0')
 
 	// Kotlin
-	implementation('org.jetbrains.kotlinx:kotlinx-metadata-jvm:0.5.0') {
+	implementation('org.jetbrains.kotlinx:kotlinx-metadata-jvm:0.6.0') {
```
Comment (modmuss50) [inline build.gradle:98]: Was this tested, I have just updated to 0.6.0 and it seems to have broken and deprecated the old API we were using?

Comment (JT122406) [reply build.gradle:98]: We can revert it but It shouldn't have broken anything

**Comment by modmuss50** [comment]
Many thanks for the PR, I have intergrated it into my PR here: https://github.com/FabricMC/fabric-loom/pull/866

The required Kotlin changes are very siginfiact, so it was just easier to do it in my PR. 👍

**Comment by JT122406** [comment]
Sounds good @modmuss50, Thank you for your time and your work here!

**Pull request closed without merging**